### PR TITLE
presigned urls take the content-type into consideration

### PR DESCRIPTION
**Pull request by kyl191**
no easy way to stop jquery from sending it

### CODE REVIEW

File: api/models/social.py [hunk 1]
```diff
@@ -229,13 +229,21 @@ def external_post(req_userid, message, media_s3, fb, tw):
 class SocialContentUploadURL(flask_restful.Resource):
     @requires_auth
     def get(self):
+        req = flask_restful.reqparse.RequestParser()
+        req.add_argument('content-type',
+                         type=str,
+                         location='args',
```
Comment (TheKevJames) [inline api/models/social.py:235]: `location='headers'`

File: api/models/social.py [hunk 2]
```diff
@@ -229,13 +229,21 @@ def external_post(req_userid, message, media_s3, fb, tw):
...
+
         userid = _request_ctx_stack.top.current_user['sub']
         suffix = ''.join(random.choice(string.ascii_uppercase + string.digits)
                          for _ in range(12))
         post_url = boto3.client('s3').generate_presigned_url(
             "put_object", {
                 "Bucket": 'calligre-images',
-                "Key": '{}-{}'.format(userid.replace('|', '-'), suffix)
+                "Key": '{}-{}'.format(userid.replace('|', '-'), suffix),
+                "ContentType": args['content-type']
```
Comment (TheKevJames) [inline api/models/social.py]: Not `Content-Type`?

File: api/models/social.py (hunk 1 above)
Comment (kyl191) [reply api/models/social.py:235]: Nope - this is the content type of the image we want to send to S3, not the content type of this request.

File: api/models/social.py (hunk 2 above)
Comment (kyl191) [reply api/models/social.py]: Also nope, the S3 client is looking for very specific naming conventions. Having a `-` in between words isn't accepted.

File: api/models/social.py (hunk 1 above)
Comment (TheKevJames) [reply api/models/social.py:235]: :+1:

File: api/models/social.py (hunk 2 above)
Comment (TheKevJames) [reply api/models/social.py]: :+1:

File: api/models/social.py [hunk 3]
```diff
@@ -229,13 +229,21 @@ def external_post(req_userid, message, media_s3, fb, tw):
 class SocialContentUploadURL(flask_restful.Resource):
     @requires_auth
     def get(self):
+        req = flask_restful.reqparse.RequestParser()
+        req.add_argument('content-type',
```
Comment (TheKevJames) [inline api/models/social.py]: Make this `Content-Type` please, then merge 👍  (and also the consumer, of course)

**Review by TheKevJames — APPROVED** [review]
Title case the `Content-Type`, then merge :+1:

File: api/models/social.py (hunk 3 above)
Comment (kyl191) [reply api/models/social.py]: Given it's in a URL, does the title case matter? Considering that we use the name as a variable.

Comment (TheKevJames) [reply api/models/social.py]: Just for consistency. Header values (ie. `Content-Type`) are traditionally title cased, rather make it the same everywhere than arbitrarily define it differently here.

**Pull request merged**